x-pack/plugins/apm/server/lib/services/get_services/get_services_items.ts

```diff
@@ -51,5 +52,6 @@ export async function getServicesItems(setup: ServicesItemsSetup) {
     ...environments,
   ];
 
-  return joinByKey(allMetrics, 'serviceName');
+  const items = joinByKey(allMetrics, 'serviceName');
+  return sortBy(items, ['serviceName']);
```

Comment:
@dgieselaar just fyi: I was getting errors on CI due to the order of items returned by `joinByKey`. Not sure if this is unstable or I'm holding it wrong. I think I've fixed it by adding an explicit sort order but might be a bandaid to an unrelated problem.

Reply:
Btw. I initially added the sorting to `joinByKey` but thought better of it. LMK if you think it belongs there.

Reply:
What about sorting in the test? The instability (in sorting) might be due to the composite aggregation.